### PR TITLE
Adding Swap Partition Support

### PR DESCRIPTION
**Pull request by MaxBrodeurUrbas**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Adding support for swap partitions diskutils.
Continuing with assumption that all partitions are primary partitions. 

###### Change Log  

- Added case for linux-swap FsType in diskutils.go

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local builds and used python scripts to stress test swap partition works as intended.

### CODE REVIEW

File: toolkit/tools/imagegen/diskutils/diskutils.go [hunk 1]
```diff
@@ -456,7 +456,7 @@ func InitializeSinglePartition(diskDevPath string, partitionNumber int, partitio
 			logger.Log.Debugf("Could not find partition path (%s). Checking other naming convention", testPartDevPath)
 		}
 		logger.Log.Warnf("Could not find any valid partition paths. Will retry up to %d times", totalAttempts)
-		err = fmt.Errorf("could not find partition to initialize in /dev")
+		err = fmt.Errorf("Could not find partition to initialize in /dev")
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/diskutils/diskutils.go:459]: This was actually correct - we've followed suggestion from the Go linters for VS Code to use small caps and no periods at the end for error messages.

**Review by PawelWMS — APPROVED** [review]
Just two small comments, looks good.

File: toolkit/tools/imagegen/diskutils/diskutils.go (hunk 1 above)
Comment (MaxBrodeurUrbas) [reply toolkit/tools/imagegen/diskutils/diskutils.go:459]: ah i see, thanks for pointing that out

**Comment by dmcilvaney** [comment]
If this is an urgent feature request, feel free to discard:

I try to add proper parsing and error checking when interacting with these types of configs. This should really be a type with well-defined values, validation, and unit tests rather than just a random string. (see `selinux.go` or `partitiontabletype.go` for example).

At minimum it would be nice to add a check to `Partition.IsValid()` that makes sure the string is one we recognize. This catches config errors early.

**Review by dmcilvaney — APPROVED** [review]

**Comment by dmcilvaney** [comment]
The refactor of the config could easily be a separate PR however.

**Comment by MaxBrodeurUrbas** [comment]
> If this is an urgent feature request, feel free to discard:
> 
> I try to add proper parsing and error checking when interacting with these types of configs. This should really be a type with well-defined values, validation, and unit tests rather than just a random string. (see `selinux.go` or `partitiontabletype.go` for example).
> 
> At minimum it would be nice to add a check to `Partition.IsValid()` that makes sure the string is one we recognize. This catches config errors early.

Thanks for the review and good point here. I will add tests for fstypes as a whole in another pr this week since none currently exist.
